app/assets/javascripts/visualizations/highvis/map.coffee

```diff
@@ -158,13 +162,14 @@ $ ->
             if (lat is null) or (lon is null)
               return
 
-            groupIndex = data.groups.indexOf dataPoint[globals.configs.groupById].toLowerCase()
-            color = globals.configs.colors[groupIndex % globals.configs.colors.length]
-
+            groupIndex = data.groups.indexOf(
+              String(dataPoint[globals.configs.groupById]).toLowerCase())
+            color = globals.getColor(groupIndex)
             latlng = new google.maps.LatLng(lat, lon)
 
             # Put aside line info if necessary
-            if @timeLines? and dataPoint[data.timeFields[0]] isnt null and not(isNaN dataPoint[data.timeFields[0]])
+            if @timeLines? and dataPoint[data.timeFields[0]] isnt null and
```

Comment:
Should this be unless (@timeLines? and dataPoint[data.timeFields[0]] or (isNaN dataPoint[data.timeFields[0]]))?

You don't have to fix this, but not isNaN looks goofy :stuck_out_tongue: 

Maybe you could do:
    and(typeof(dataPoint[data.timeFields[0]]) is 'number')